bcos-txpool/bcos-txpool/txpool/storage/MemoryStorage.cpp

```diff
@@ -301,6 +317,11 @@ TransactionStatus MemoryStorage::insertWithoutLock(Transaction::Ptr transaction)
     auto [it, inserted] = m_txsTable.insert(std::make_pair(transaction->hash(), transaction));
     if (!inserted)
     {
+        if (transaction->submitCallback() && !it->second->submitCallback())
+        {
+            it->second->setSubmitCallback(std::move(transaction->submitCallback()));
+            return TransactionStatus::None;
+        }
```

Comment:
add filter logic when rpc insert slow then p2p insert. rpc insert will have submit callback and p2p dont.